### DISCUSSION TITLE
Linux: "BASS faulted with error code 3: Driver." on startup. No audio.

### DISCUSSION THREAD

**Discussion by IgorGRBR** [ppy/osu#26974] [Q&A] [answered]
### Type

Game behaviour

### Bug description

No audio since the recent osu!lazer update. The game also displays "BASS faulted with error code 3: Driver" notifications upon startup.

### Screenshots or videos

![Screenshot_20240202_202722](https://github.com/ppy/osu/assets/25769098/4d781b6a-f837-4656-93b0-b31b47fb7add)

### Version

2024.131.0-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14144472/compressed-logs.zip)

**Comment by bdach** [comment]
There were no major changes to audio since last release so I'd say it's more likely to be an issue with your system.

Please check if rolling back to an earlier version does anything. If I'm right, you should still be seeing the error on old releases.

**Reply by IgorGRBR** [reply]
It seems that the problem is indeed related to my system, as rolling back did not help. Do you happen to know what system lib might be causing it? Can I try to run the game with an older version of the problematic library?

**Reply by bdach** [reply]
It's probably whatever you have that is running audio (alsa or pulseaudio or pipewire). I'd try switching the default audio device in game settings and rolling back any recent update of relevant audio packages if the former does not help.

**Reply by IgorGRBR** [reply]
The audio device in settings was already at default. I've discovered that if I switch it to whatever device I want to play audio through, the audio starts playing, but any other program running is unable to play audio through that device. And vice-versa - if another program is already playing the audio through the selected device, osu! is unable to play audio through that device no matter what. Rolling back packages is, unfortunately, not an option for the distro I use (or I'm too dumb to figure that out at least).

Either way, thank you for the help!

**Comment by hwsmm** [comment]
This is probably due to this [PipeWire issue](https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/3830). It's already fixed in 1.0.3, so I guess you can just wait for it to reach your distro repository?

**Reply by IgorGRBR** [reply]
I see. I'm indeed using pipewire version 1.0.2. I will wait for an update and report it here if it fixes this issue.

**Reply by JoseGuilhermeCR** [reply]
For now, downgrading fixed it for me. On Fedora `dnf downgrade pipewire` and a reboot should be enough.

**Comment by IgorGRBR** [comment, marked as answer]
Update: pipewire v1.0.3 fixes this issue, so this is no longer a problem.

**Reply by Alex-162** [reply]
Just ran a system upgrade 5 minutes ago, pipewire is on v1.0.3 for me, and still no audio. Any suggestions?

**Reply by JoseGuilhermeCR** [reply]
Did you reboot? @Alex-162

**Reply by Alex-162** [reply]
of course @JoseGuilhermeCR

**Reply by JoseGuilhermeCR** [reply]
Well, then I don't know what might've happened. I'll leave the list of packages that were updated together with pipewire in my system below, because I don't know if any of them might've caused any effect.... but other than that, good luck.

```
    Upgrade  alsa-lib-1.2.11-2.fc39.x86_64                                @updates
    Upgraded alsa-lib-1.2.10-3.fc39.x86_64                                @@System
    Upgrade  alsa-ucm-1.2.11-2.fc39.noarch                                @updates
    Upgraded alsa-ucm-1.2.10-3.fc39.noarch                                @@System
    Upgrade  alsa-utils-1.2.11-1.fc39.x86_64                              @updates
    Upgraded alsa-utils-1.2.10-1.fc39.x86_64                              @@System
    Upgrade  pipewire-1.0.3-1.fc39.x86_64                                 @updates
    Upgraded pipewire-0.3.80-1.fc39.x86_64                                @@System
    Upgrade  pipewire-alsa-1.0.3-1.fc39.x86_64                            @updates
    Upgraded pipewire-alsa-0.3.80-1.fc39.x86_64                           @@System
    Upgrade  pipewire-gstreamer-1.0.3-1.fc39.x86_64                       @updates
    Upgraded pipewire-gstreamer-0.3.80-1.fc39.x86_64                      @@System
    Upgrade  pipewire-jack-audio-connection-kit-1.0.3-1.fc39.x86_64       @updates
    Upgraded pipewire-jack-audio-connection-kit-0.3.80-1.fc39.x86_64      @@System
    Upgrade  pipewire-jack-audio-connection-kit-libs-1.0.3-1.fc39.x86_64  @updates
    Upgraded pipewire-jack-audio-connection-kit-libs-0.3.80-1.fc39.x86_64 @@System
    Upgrade  pipewire-libs-1.0.3-1.fc39.x86_64                            @updates
    Upgraded pipewire-libs-0.3.80-1.fc39.x86_64                           @@System
    Upgrade  pipewire-pulseaudio-1.0.3-1.fc39.x86_64                      @updates
    Upgraded pipewire-pulseaudio-0.3.80-1.fc39.x86_64                     @@System
    Upgrade  pipewire-utils-1.0.3-1.fc39.x86_64                           @updates
    Upgraded pipewire-utils-0.3.80-1.fc39.x86_64                          @@System
    Upgrade  pipewire-codec-aptx-1.0.2-1.fc39.x86_64                      @rpmfusion-free-updates
    Upgraded pipewire-codec-aptx-0.3.80-1.fc39.x86_64                     @@System
```

**Reply by Alex-162** [reply]
figured it out... recently had to restore my files from a backup, somehow the ALSA drivers got lost in that, just happened to line up with this issue, bad timing lol. putting those back in worked.

**Comment by crispy-caesus** [comment]
while my audio seems to be working, I still get this error

**Reply by crispy-caesus** [reply]
oh, needed pipewire alsa

**Reply by tensaro** [reply]
I can confirm, I have pipewire 1.4.2 and after installing pipewire-alsa everything works fine
